controllers/volumereplicationgroup_controller.go

```diff
@@ -1145,6 +1184,74 @@ func (v *VRGInstance) createVR(vrNamespacedName types.NamespacedName, state volr
 	return nil
 }
 
+// namespacedName applies to both VolumeReplication resource and pvc as of now.
+// This is because, VolumeReplication resource for a pvc that is created by the
+// VolumeReplicationGroup has the same name as pvc. But in future if it changes
+// functions to be changed would be processVRAsPrimary(), processVRAsSecondary()
+// to either receive pvc NamespacedName or pvc itself as an additional argument.
+func (v *VRGInstance) selectVolumeReplicationClass(namespacedName types.NamespacedName) (string, error) {
+	className := ""
+
+	if len(v.replClassList.Items) == 0 {
+		v.log.Info("No VolumeReplicationClass available")
+
+		return className, fmt.Errorf("no VolumeReplicationClass available")
+	}
+
+	pvc := &corev1.PersistentVolumeClaim{}
+	if err := v.reconciler.Get(v.ctx, namespacedName, pvc); err != nil {
```

Comment:
We already listed the PVCs and should not be "getting" them again. Reduce APi calls, and reuse fetched PVCs instead here (or in a higher function and pass the required replClass to lower functions, from an abstraction POV)